sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorClient/MetricsLiveTests.cs

```diff
@@ -0,0 +1,172 @@
+                }
+            }
+
+            Assert.That(definitionCount, Is.GreaterThan(0));
+        }
+
+        [Test]
+        public async Task GetMetricSeriesData()
+        {
+            MetricsAdvisorClient client = GetMetricsAdvisorClient();
+
+            var seriesKey1 = new DimensionKey();
+            seriesKey1.AddDimensionColumn("city", "Delhi");
+            seriesKey1.AddDimensionColumn("category", "Handmade");
+
+            var seriesKey2 = new DimensionKey();
+            seriesKey2.AddDimensionColumn("city", "Koltaka");
+            seriesKey2.AddDimensionColumn("category", "__SUM__");
+
+            var seriesKeys = new List<DimensionKey>() { seriesKey1, seriesKey2 };
+            var returnedKeys = new List<DimensionKey>();
+
+            var options = new GetMetricSeriesDataOptions(seriesKeys, SamplingStartTime, SamplingEndTime);
+
+            await foreach (MetricSeriesData seriesData in client.GetMetricSeriesDataAsync(MetricId, options))
+            {
+                Assert.That(seriesData, Is.Not.Null);
+                Assert.That(seriesData.Definition, Is.Not.Null);
+                Assert.That(seriesData.Definition.SeriesKey, Is.Not.Null);
+                Assert.That(seriesData.Timestamps, Is.Not.Null);
+                Assert.That(seriesData.Values, Is.Not.Null);
+
+                Assert.That(seriesData.Definition.MetricId, Is.EqualTo(MetricId));
+
+                Assert.That(seriesData.Timestamps.Count, Is.EqualTo(seriesData.Values.Count));
+
+                foreach (DateTimeOffset timestamp in seriesData.Timestamps)
+                {
+                    Assert.That(timestamp, Is.InRange(SamplingStartTime, SamplingEndTime));
+                }
+
+                returnedKeys.Add(seriesData.Definition.SeriesKey);
+            }
+
+            IEnumerable<List<KeyValuePair<string, string>>> expectedKvps = seriesKeys.Select(key => key.AsDictionary().ToList());
+            IEnumerable<List<KeyValuePair<string, string>>> returnedKvps = returnedKeys.Select(key => key.AsDictionary().ToList());
+
+            Assert.That(returnedKvps, Is.EquivalentTo(expectedKvps));
+        }
+
+        [Test]
+        public async Task GetMetricEnrichmentStatuses()
+        {
+            MetricsAdvisorClient client = GetMetricsAdvisorClient();
+
+            var options = new GetMetricEnrichmentStatusesOptions(SamplingStartTime, SamplingEndTime);
+
+            var statusCount = 0;
+
+            await foreach (EnrichmentStatus enrichmentStatus in client.GetMetricEnrichmentStatusesAsync(MetricId, options))
+            {
+                Assert.That(enrichmentStatus, Is.Not.Null);
+                Assert.That(enrichmentStatus.Status, Is.Not.Null.And.Not.Empty);
+                Assert.That(enrichmentStatus.Message, Is.Not.Null.And.Not.Empty);
+                Assert.That(enrichmentStatus.Timestamp, Is.InRange(SamplingStartTime, SamplingEndTime));
+
+                if (++statusCount >= MaximumSamplesCount)
```

Comment:
Just curious, before the tests were fetching 1 to 2 data points and then breaking once there was validation for 2 points. Looking at `MaximumSamplesCount=10`. Any reason for the increase?

Reply:
Just wanted to get more samples for validation. The tests we had previously were only basic placeholders to make sure we didn't have any major bugs in the SDK.

Reply:
Will the data coming back be so different than 2 vs 10 can caught a problem?

Reply:
Maybe not for every endpoint. We were able to stumble upon an issue that only happened sometimes, though (https://github.com/Azure/azure-sdk-for-net/issues/16581). The service bug is only triggered when an incident with `Status = Resolved` is returned.

We have two possible overloads for listing incidents. One of them was able to detect this issue, but the other one only got back active incidents and couldn't trigger it (even with the cap of 10 samples). This was totally by chance, though. We can't do much to guarantee we're testing all possible service responses.

Another reason is that we have some abstract types with multiple concrete implementations the service can return (NotificationHook => WebNotificationHook, EmailNotificationHook; MetricFeedback => AnomalyMetricFeedback; ChangePointFeedback; ...), and having a large set of samples makes it more likely that we'll be testing at least some of them. This depends on the method we're testing, though, so we could set a different amount of samples depending on the test.

I agree with you that it's a question about finding a good balance (1000 would work, but it's too much). I don't think 10 samples would have a big impact on test performance, though.